### PR TITLE
New feature: gin create --here

### PR DESCRIPTION
**Pull request by achilleas-k**
## New feature: Create repository from current directory

Running `gin create --here` will create a repository on the server and initialise the current directory, setting defaults and the new repository as the default remote.

## Additional changes

Minor fixes to release script.

### CODE REVIEW

File: gin-client/repos.go [hunk 1]
```diff
@@ -205,18 +205,37 @@ func (gincl *Client) CloneRepo(repoPath string) (string, error) {
...
 	if err != nil {
 		return "", err
 	}
 	green.Println("OK")
+	_, repoName := splitRepoParts(repoPath)
+	Workingdir = repoName
+	return gincl.InitDir(repoPath)
+}
 
+// InitDir initialises the local directory with
```
Comment (mpsonntag) [inline gin-client/repos.go]: Incomplete docstring

File: gin-client/repos.go [hunk 2]
```diff
@@ -205,18 +205,37 @@ func (gincl *Client) CloneRepo(repoPath string) (string, error) {
...
+	if !IsRepo() {
+		stdout, stderr, err := RunGitCommand("init")
+		if err != nil {
+			util.LogWrite("Error during Init command")
+			util.LogWrite("[stdout]\r\n%s", stdout.String())
+			util.LogWrite("[stderr]\r\n%s", stderr.String())
+			return "", initerr
+		}
+		Workingdir = "."
+	} else {
```
Comment (mpsonntag) [inline gin-client/repos.go]: Why the empty `else` clause here?

File: gin-client/repos.go [hunk 3]
```diff
@@ -250,15 +269,33 @@ func (gincl *Client) CloneRepo(repoPath string) (string, error) {
 		return "", err
 	}
 
+	err = AddRemote("origin", remotePath)
+	// Ignore if it already exists
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return "", initerr
+	}
+
+	if new {
```
Comment (mpsonntag) [inline gin-client/repos.go]: Could this `if new` be combined with the `if new` at L290 or should they remain separate?

**Comment by achilleas-k** [comment]
Thanks for the comments. They were all sloppy mistakes.

**Pull request merged**